cmd/deploy.go

```diff
@@ -181,36 +181,41 @@ func autofillApimodel(dc *deployCmd) {
 	if !useManagedIdentity {
 		spp := dc.containerService.Properties.ServicePrincipalProfile
 		if spp != nil && !(spp.ClientID == "" && spp.Secret == "") {
-			log.Fatal("apimodel invalid: ServicePrincipalProfile is missing either the clientid or secret.")
-		}
+			log.Info("using user supplied ServicePrincipalProfile in apimodel.")
 
-		log.Warnln("apimodel: ServicePrincipalProfile was empty, creating application...")
+			dc.containerService.Properties.ServicePrincipalProfile = &api.ServicePrincipalProfile{
```

Comment:
The point of the conditional on line 183 is to check if only ONE of the two values is set. It looks like it one of my internal refactors when writing the regressing PR removed the original check. It was originally:

- if either is not set {
   - if one is set { throw a more specific error }
- }